### ISSUE TITLE
SSL Validation Fails for New Certificates from Let's Encrypt

### ISSUE THREAD

**Issue by mkelly1495** [pact-foundation/pact-net#330]
I am using `PactNet 3.0.0` on windows (developer machine) and linux (jenkins pipeline). Our project just updated the certificate used for our internal pact-broker server because the old one expired on Sept-30. Now if I try to run any of our verification tests, the test fails with `SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed`. I did a bit of digging and determined that setting the `SSL_CERT_FILE` environment variable resolves the issue when using a newer version of ruby (e.g. 3.0). But the standalone version of ruby `ruby 2.2.2p95` distributed with PactNet 3.0.0 does not appear to use the `SSL_CERT_FILE` environment variable at all. In fact, when I run the [SSl Doctor script](https://github.com/mislav/ssl-tools), it indicates that the SSL path being used is actually hard coded to `C:/Users/Justin/Projects/knap-build/var/knapsack/software/x86-windows/openssl/1.0.1l/ssl/certs/` (which doesn't exist). The `SSL_CERT_FILE` and `SSL_CERT_DIR` environment variables are totally ignored, so setting them has no effect. As of now, there is no work-around for this issue since the problem appears to be caused by the stand-alone ruby version included with PactNet.Windows and PactNet.Linux. I have verified that the latest version of the [pact-ruby-standalone](https://github.com/pact-foundation/pact-ruby-standalone/releases/tag/v1.88.76) does not have the same SSL verification issue. However, I don't know if simply upgrading to the latest will break PactNet. Any ideas?

BTW - here is the output from doctor.rb (executed using the pact-ruby-standalone included with PactNet 3.0.0) when attempting to access github.com:443 on a windows box (note that I have
```
$ export SSL_CERT_FILE="D:/projects/certs"
$ ./ruby.bat /d/Projects/ssl-tools/doctor.rb github.com:443

OpenSSL 1.0.1l 15 Jan 2015: C:/Users/Justin/Projects/knap-build/var/knapsack/software/x86-windows/openssl/1.0.1l/ssl
SSL_CERT_DIR=""
SSL_CERT_FILE="D:\\Projects\\MyProject\\Provider\\bin\\Debug\\netcoreapp3.1\\pact-win32\\lib\\ruby\\bin\\..\\lib\\ca-bundle.crt"

HEAD https://github.com:443
OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed

The server presented a certificate that could not be verified:
  subject: /C=US/O=DigiCert, Inc./CN=DigiCert High Assurance TLS Hybrid ECC SHA256 2020 CA1
  issuer: /C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert High Assurance EV Root CA
  error code 20: unable to get local issuer certificate

Possible causes:
  `C:/Users/Justin/Projects/knap-build/var/knapsack/software/x86-windows/openssl/1.0.1l/ssl/certs/' is empty
```

**Comment by bethesque** [comment]
See https://github.com/pact-foundation/pact-ruby-standalone/issues/69

We'll be attempting to update the bundled certs in Travelling Ruby. In the meantime, try setting `PACT_DISABLE_SSL_VERIFICATION=true` or set a custom `SSL_CERT_FILE` using the certificate chain exported from your broker. https://docs.pact.io/pact_broker/advanced_topics/using-tls/#extracting-a-copy-of-the-certificate-from-a-running-pact-broker

**Comment by mkelly1495** [comment]
It looks like settinging `PACT_DISABLE_SSL_VERIFICATION=true` has no effect. Also, as I mentioned, the `SSL_CERT_FILE` is totally ignored by the `1.88.14` (included with PactNet 3.0.0) version of the `pact-ruby-standalone` (so setting that environment variable has no effect).

**Comment by mkelly1495** [comment]
I've been doing a bit more testing and it appears that the `1.88.76` version of the pact-ruby-standalone works fine accessing github:443, but I still get the ssl verification error when attempting to access my pact-broker site (cert signed with Let's Encrypt). Based on the `doctor.rb` script, it looks like the `SSL_CERT_FILE` is still being ignored by the `1.88.76` version of pact-ruby-standalone.

**Comment by mefellows** [comment]
Please see the latest release (`3.0.1`) which now has the updated embedded ruby standalone. Release is currently being [baked now](https://ci.appveyor.com/project/PactFoundation/pact-net) 🧑‍🍳  but artifacts are on the way.

Note that this won't fully resolve the issue, you will still need to set env vars temporarily until the upstream problem is fixed: https://github.com/pact-foundation/pact-net/issues/330#issuecomment-933821154

**Comment by mkelly1495** [comment]
> For which commands is it being ignored? There's a working example here for can-i-deploy https://github.com/DiUS/pact_broker-docker/blob/master/docker-compose.yml

Try running the [doctor.rb](https://github.com/mislav/ssl-tools) script using the `1.88.76` version of the traveling ruby. The output indicates that the `SSL_CERT_DIR` environment variable appears to be hardcoded to `...\\netcoreapp3.1\\pact-win32\\lib\\ruby\\bin\\..\\lib\\ca-bundle.crt`.

**Comment by mkelly1495** [comment]
The easiest fix for the problem is probably to just replace the `[platform]/lib/ruby/lib/ca-bundle.crt` with a valid certificate (current one is expired). I replaced the existing cert with the cert distributed with mingw64 and the issue was resolved.

**Comment by mkelly1495** [comment]
On windows, PactNet overwrites the `SSL_CERT_FILE` with the value in `pact-win32/config.json`. It is also overwritten by the `pact-win32/lib/ruby/bin/ruby.bat` script.

**Comment by adamrodger** [comment]
I believe this is fixed in v3.0.2

**Comment by DanBiscotti** [comment]
I'm currently getting this issue on v3.0.2 is it possible the cert has expired again?
